ablySpec/RealtimeClientChannel.swift

```diff
@@ -609,6 +609,88 @@ class RealtimeClientChannel: QuickSpec {
 
             }
 
+            // RTL8
+            context("unsubscribe") {
+
+                // RTL8a
+                it("with no arguments unsubscribes the provided listener to all messages if subscribed") {
+                    let options = AblyTests.commonAppSetup()
+                    let client1 = ARTRealtime(options: options)
+                    defer { client1.close() }
+                    let client2 = ARTRealtime(options: options)
+                    defer { client2.close() }
+
+                    let channel1 = client1.channels.get("test")
+                    let channel2 = client2.channels.get("test")
+
+                    class Test {
+                        static var counter = 0
+                        private init() {}
+                    }
+
+                    channel1.subscribe { message in
+                        expect(message.data as? String).to(equal("message"))
+                        Test.counter += 1
+
+                        channel1.unsubscribe()
+                    }
+
+                    channel2.subscribe { message in
+                        expect(message.data as? String).to(equal("message"))
+                        Test.counter += 1
+                    }
+
+                    channel1.publish(nil, data: "message")
+                    channel1.publish(nil, data: "message")
+
+                    expect(Test.counter).toEventually(equal(3), timeout: testTimeout)
```

Comment:
I really don't like the readability of the test.  Why not simply set up one channel, subscribe, publish and check that message is received. Then unsubscribe and check that no message is received?  This is overly complex and I don't really follow why one channel unsubscribes whilst the other one doesn't.